Sources/AudioKit/Nodes/Playback/AudioPlayer/AudioPlayer.swift

```diff
@@ -97,11 +122,15 @@ public class AudioPlayer: Node {
     // MARK: - Internal functions
 
     func internalCompletionHandler() {
-        guard isPlaying, engine?.isInManualRenderingMode == false else { return }
+        guard isPlaying, !isPaused, engine?.isInManualRenderingMode == false else { return }
 
         scheduleTime = nil
-        completionHandler?()
         isPlaying = false
+        elapsedTimeOffset = 0
+
+        DispatchQueue.main.async { [weak self] in
+            self?.completionHandler?()
```

Comment:
Is this something worth backporting until this PR is fixed? (or any other fixes in here for that matter)